avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
+                               sha.sha(fingerprint).hexdigest()[:4])
```

Comment:
Consider a qemu boot test with custom command being executed (based on params). In one variant you execute `lspci`, in other `dmidecode` and then you have several variants of the qemu configuration.

The variant-independent data can be recorded in the main `datadir` and the per-variant data would indeed be per-variant as people can use multiple different files, or even inject some values.

Basically you can find examples which would benefit in only node names being used, then variants where full node-paths and for some examples the whole content is important. There is no perfect solution as it depends on the usecase. My view on this is that when the value changes (either by a different yaml file, or via --mux-inject or whatever...) it's a different variant.